lib/neo4j/core/cypher_session/responses/bolt.rb

```diff
@@ -52,10 +52,16 @@ def wrap_entity(entity_data)
 
           def extract_message_groups(flush_messages_proc)
             fields_messages = flush_messages_proc.call
+
             validate_message_type!(fields_messages[0], :success)
 
             result_messages = []
-            while (messages = flush_messages_proc.call)[0].type != :success
+            messages = nil
+
+            loop do
+              messages = flush_messages_proc.call
+              next if messages.nil?
```

Comment:
Interesting....  I think we must get this when the server returns something in the socket but without any data.  Otherwise I think we would just sit listening to the socket and eventually time out.

So I *think* that this is fine.  If it works in the specs and if it works in your app then we should merge it and I'll try it in our app.